apps-rendering/src/components/Callout/formFields.tsx

```diff
@@ -0,0 +1,148 @@
+import type { FormField as formFieldType } from '@guardian/apps-rendering-api-models/formField';
+import {
+	Option,
+	Select,
+	TextArea,
+	TextInput,
+} from '@guardian/source-react-components';
+import { FileInput } from '@guardian/source-react-components-development-kitchen';
+import CheckboxInput from 'components/CheckboxInput';
+import RadioInput from 'components/RadioInput';
+import type { FC } from 'react';
+import { logger } from '../../logger';
+import type { FormDataType, ValidationErrors } from './calloutForm';
+import { fieldInput, textarea } from './styles';
+
+type FormFieldProp = {
+	formId: number;
+	formField: formFieldType;
+	formData: FormDataType;
+	setFieldInFormData: (
+		id: string,
+		data: string | string[] | undefined,
+	) => void;
+	validationErrors: ValidationErrors;
+};
+
+export const FormField: FC<FormFieldProp> = ({
+	formId,
+	formField,
+	formData,
+	setFieldInFormData,
+	validationErrors,
+}) => {
+	const { type, label, description, mandatory, options, id } = formField;
+	const name = `field_${type}_${id}`;
+	const fieldValue =
+		formField.id in formData ? (formData[formField.id] as string) : '';
+	const fieldError = validationErrors[formField.id];
+
+	const selectOptions = options.map(({ value, label }) => {
+		return (
+			<Option key={value} value={value}>
+				{label}
+			</Option>
+		);
+	});
+	if (!fieldValue || !mandatory)
+		selectOptions.unshift(
+			<Option value="">Please select an option</Option>,
+		);
+
+	switch (type) {
+		case 'text':
+		case 'number':
+		case 'phone':
```

Comment:
In the case of `number` or `phone` you can set `inputMode` to `decimal`. Then, both Android and iOS will show a number pad. https://medium.com/web-dev-survey-from-kyoto/how-to-handle-invalid-user-inputs-in-react-forms-for-ux-design-best-practices-e3108ef8a793

Reply:
Thanks for the link to the docs. I can't quite figure out if it's saying that `type` is going to also set the browser keyboard or not (just like `inputMode` does). It's a bit hard to test as you have to do it on device I guess. 

It sounds to me like if you want to actually change what keyboard is displayed (for a better user experience etc) you'd have to also set the `inputMode` but I may be wrong. Either way, all good - we can test and make this minor improvement when live. https://www.c-sharpcorner.com/article/inputmode-and-type-in-html-inputs-for-better-user-experiences/